backend/contactsAPI.py

```diff
@@ -73,14 +73,110 @@ def create_user():
+                                  "emails": [],
+                                  "phones": [],
+                                  "labels": [],
+                                  "image": {"type": "none", "url": ""}})
+        for i, field in enumerate(fields):
+            if i == len(contact):
+                break
+            if field == "Name" and contact[i]:
+                new_db_contact["name"] = contact[i]
+
+            elif "E-mail" in field and "Type" in field and contact[i]:
+                new_db_contact["emails"].append({"address": contact[i + 1],
+                                                 "type": contact[i].strip("* ")})
+            elif "Phone" in field and "Type" in field and contact[i]:
+                new_db_contact["phones"].append({"number": contact[i + 1],
+                                                 "type": contact[i].strip("* ")})
+            elif "Group Membership" in field and contact[i]:
+                new_db_contact["labels"] = contact[i].split(" ::: ")
+                new_db_contact["labels"] = [label.strip("* ") for label in new_db_contact["labels"]]
+
+            elif "Photo" in field and contact[i]:
+                new_db_contact["image"] = {"type": "external", "url" : contact[i]}
+        new_db_contacts.append(new_db_contact)
+
+    return new_db_contacts
+
+def parse_outlook_csv(fields, contacts):
+    new_db_contacts = []
+    for contact in contacts: 
+        new_db_contact = Contacts({"name": "", 
+                                  "emails": [],
+                                  "phones": [],
+                                  "labels": [],
+                                  "image": {"type":"none", "url": ""}})
+        for i, field in enumerate(fields):
+            if i == len(contact):
+                break
+            if "Name" in field and contact[i]:
+                new_db_contact["name"] += contact[i]
+
+            elif "E-mail" in field and contact[i]:
+                new_db_contact["emails"].append({"address": contact[i],
+                                                 "type": str(len(new_db_contact["emails"]) + 1)})
+            elif "Phone" in field and contact[i]:
+                new_db_contact["phones"].append({"number": contact[i],
+                                                 "type": field.replace(" Phone", "")})
+            elif "Categories" in field and contact[i]:
+                while i < len(contact) and contact[i]:
+                    new_db_contact["labels"].append(contact[i])
+                    i += 1
+
+            #outlook csv's don't have any kind of photo field
+
+        #csv_util_clear_empty_fields(new_db_contact)
+        new_db_contacts.append(new_db_contact)
+    
+    return new_db_contacts
+
+def parse_upload_csv(token, fields, contacts):
+    contacts = [[entry.strip() for entry in contact.decode().split(',')] for contact in contacts]
+    user = User().find_by_token(token)
+    if fields[0] == "Name":
```

Comment:
Is there any other way to determine which contacts csv we are getting?

Reply:
This is definitely the easiest method I found, and always works to separate the two.